### ISSUE TITLE
ComboBox defaultSelectedKey doesn't work within BrowserRouter

### ISSUE THREAD

**Issue by cobbs-totem** [OfficeDev/office-ui-fabric-react#3185]
I could not get the defaultSelectedKey to display the default value in my ComboBox, and after some trial and error, I determined that it simply would not work, if I rendered the component inside a BrowserRouter (react-router-dom).  

I'm not sure how BrowserRouter works when it renders it's children, but there's a disconnect somewhere.

**Comment by cobbs-totem** [comment]
@erichdev So, this is really hard to isolate..  I no longer believe it has anything to do with BrowserRouter, after much more testing.  My app uses redux and when it mounts, does some async calls which causes a few state changes.

What I *think* happens is that the state change causes the ComboBox to get new props, which determines that props.selectedKey is no longer equal to this.props.selectedKey (which I never pass, I only set the defaultSelectedKey.  This results in selectedKey being set to undefined.

I think I worked around this by setting both the defaultSelectedKey and selectedKey.
